src/Moryx/Bindings/TextBindingResolverFactory.cs

```diff
@@ -14,7 +14,7 @@ public static class TextBindingResolverFactory
         /// Regex for binding.
         /// It matches strings of type "Product.Id" or "Recipe".
         /// </summary>
-        private const string BindingRegex = @"\{(?<binding>[A-Z][a-z]+(\.?[A-Z]\w+(?:\[\w+\])?)*)(?<hasFormat>:(?<format>\w+))?\}";
+        private const string BindingRegex = @"\{(?<binding>[A-Z][a-z][0-9]+(\.?[A-Z]\w+(?:\[\w+\])?)*)(?<hasFormat>:(?<format>\w+))?\}";
```

Comment:
I think this is what you were aiming for
```
@"\{(?<binding>([A-Za-z_][A-Za-z0-9_]*)(\[\d+\])?(\.([A-Za-z_][A-Za-z0-9_]*)(\[\d+\])?)*)?(?<hasFormat>:(?<format>\w+))?\}"
```
Your suggestion would only match anything starting with capital letter, followed by lower letter and a set of numbers, e.g. `Co1`. I took the freedom to build upon your suggestion and allow for any valid C# Property names, followed by the brackets and formating already allowed before 😊